### PR TITLE
Renamed DroneCore to DronecodeSDK

### PR DESCRIPTION
**Pull request by julianoes**
It was decided that DroneCore should be renamed to Dronecode SDK because
that's essentially what it is and should create less confusion.

This commit renames:
- The DroneCore class to DronecodeSDK.
- The dronecore library to dronecode_sdk.
- The dronecore.h|cpp source files.
- Some remaining instances of DroneCore in the README.
- The proto submodule with renamed package names.
- Various install and CI paths.
- The dockerhub repository names.

It does not rename the github repository URLs yet since the repository
has not been moved/renamed yet.

@JonasVautherin please review the changes in backend in particular.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ The library provides both synchronous (blocking) API calls, as well as asynchron
 
 ## API Overview
 
-API consumers use the `DroneCore` class to discover and manage vehicles (`System` objects). Using the `System` object plugins such as e.g. `Action`, `Telemetry`, or `Mission` can be instantiated which provide information about the state of the drone and allow to interact with it.
+API consumers use the `DronecodeSDK` class to discover and manage vehicles (`System` objects). Using the `System` object plugins such as e.g. `Action`, `Telemetry`, or `Mission` can be instantiated which provide information about the state of the drone and allow to interact with it.
 
 The links below take you to the respective header files:
```
Comment (hamishwillee) [inline README.md]: Link below needs to be fixed: [dronecore](include/dronecore.h)

File: core/dronecode_sdk.h [hunk 2]
```diff
@@ -8,20 +8,21 @@
...
-namespace dronecore {
+namespace dronecode_sdk {
 
-class DroneCoreImpl;
+class DronecodeSDKImpl;
 class System;
 
 /**
- * @brief This is the main class of **%DroneCore MAVLink API Library** (for the Dronecode Platform).
+ * @brief This is the main class of **%DronecodeSDK MAVLink API Library** (for the Dronecode
```
Comment (hamishwillee) [inline core/dronecode_sdk.h]: Maybe change to:

> This is the main class of **Dronecode SDK** (a MAVLink API Library for the Dronecode Platform).

File: core/plugin_impl_base.h [hunk 3]
```diff
@@ -17,14 +17,14 @@ class PluginImplBase {
      * when a system is constructed. This does not mean that the system actually
      * exists and is connected, it might just be an empty dummy system.
      *
-     * Plugins should do initialization steps with other parts of DroneCore
+     * Plugins should do initialization steps with other parts of DronecodeSDK
```
Comment (hamishwillee) [inline core/plugin_impl_base.h]: "... other parts of Dronecode SDK" - This is a reference to the library, not the class.

File: example/fly_mission/fly_mission.cpp [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /**
  * @file fly_mission.cpp
  *
- * @brief Demonstrates how to Add & Fly Waypoint missions using DroneCore.
+ * @brief Demonstrates how to Add & Fly Waypoint missions using DronecodeSDK.
```
Comment (hamishwillee) [inline example/fly_mission/fly_mission.cpp]: using the Dronecode SDK

**Comment by hamishwillee** [comment]
Wow. I've scanned through this and fixed up all but the issue above (link to dronecore.h in the README). The changes were generally just DronecodeSDK to "the Dronecode SDK" in the cases where we were talking about the SDK and not the class.

Updating the docs will be a challenge.

**Comment by julianoes** [comment]
Thanks a lot @hamishwillee!

File: README.md (hunk 1 above)
Comment (julianoes) [reply README.md]: Not yet, as written in the description we should change them once the github repo has moved.

**Pull request merged**